cmd/cleanup-jobs.go

```diff
@@ -36,11 +40,17 @@ var CleanupJobs = &cobra.Command{
 
 			go func(j v1.Job, clientSet *kubernetes.Clientset) {
 
-				// if job status is not equal to "succeeded", delete the job
-				if j.Status.Failed == 0 {
-					log.Printf("INFO: Removing job %v from namespace %v", j.Name, j.Namespace)
-					err = clientSet.BatchV1().Jobs(namespace).Delete(context.TODO(), j.Name, metav1.DeleteOptions{})
-					check(err)
+				// loop through the job object's status conditions
+				for _, conditions := range j.Status.Conditions {
+
+					// if the type of the JobCondition is equal to "Failed", delete the job
+					if conditions.Type == "Failed" {
+						log.Infof("Removing failed job %v from namespace %v. Failed reason: %v", j.Name, j.Namespace, conditions.Reason)
+						if err = clientSet.BatchV1().Jobs(namespace).Delete(context.TODO(), j.Name, metav1.DeleteOptions{}); err != nil {
+							log.Fatalf("Failed to delete job: %v", err)
```

Comment:
This should be Errorf() rather than Fatalf - rather attempt to delete all failed jobs and report which could or could not be deleted than hard exit in the middle of a concurrent loop.